src/content-tags-drawer/api/hooks/selectors.js

```diff
@@ -0,0 +1,56 @@
+// @ts-check
+import { useTaxonomyTagsData, useContentTaxonomyTagsData, useContentData } from './api';
+
+/**
+ * @returns {import("../types.mjs").TaxonomyTagsData | undefined}
+ */
+export const useTaxonomyTagsDataResponse = (taxonomyId, fullPathProvided) => {
+  const response = useTaxonomyTagsData(taxonomyId, fullPathProvided);
+  if (response.status === 'success') {
+    return response.data.data;
+  }
+  return undefined;
+};
+
+/**
+ * @returns {boolean}
+ */
+export const useIsTaxonomyTagsDataLoaded = (taxonomyId, fullPathProvided) => (
+  useTaxonomyTagsData(taxonomyId, fullPathProvided).status === 'success'
+);
+
+/**
+ * @returns {import("../types.mjs").ContentTaxonomyTagsData | undefined}
+ */
+export const useContentTaxonomyTagsDataResponse = (contentId) => {
+  const response = useContentTaxonomyTagsData(contentId);
+  if (response.status === 'success') {
+    return response.data;
+  }
+  return undefined;
+};
+
+/**
+ * @returns {boolean}
+ */
+export const useIsContentTaxonomyTagsDataLoaded = (contentId) => (
+  useContentTaxonomyTagsData(contentId).status === 'success'
+);
+
+/**
+ * @returns {import("../types.mjs").ContentData | undefined}
+ */
+export const useContentDataResponse = (contentId) => {
+  const response = useContentData(contentId);
+  if (response.status === 'success') {
+    return response.data.data;
+  }
+  return undefined;
+};
+
+/**
+ * @returns {boolean}
+ */
+export const useIsContentDataLoaded = (contentId) => (
+  useContentData(contentId).status === 'success'
+);
```

Comment:
I'll echo the same concerns I showed [here](https://github.com/openedx/frontend-app-course-authoring/pull/645#discussion_r1377342836).

If a function starts with use is it treated as a react hook by the linters. None of these need to be react hooks, so the "use" at the beginning can be dropped.

Reply:
I could be mistaken, but since the functions in `/api/hooks/api.js` return `useQuery` which is a hook, they also become hooks. If we drop the "use" in the functions in the `/api/hooks/selectors.js` that call them, I get linter errors saying that the hooks cannot be called inside normal functions and suggests to either convert them to hooks (the current case with "use") or convert them to React component functions which doesn't seem right? Maybe I am misunderstanding how to go about addressing this.

Reply:
My understanding was the same as @yusuf-musleh's - any function which uses a hook is itself a hook (or a component) so should be named `use...`.

Reply:
@xitij2000 @bradenmacdonald I changed the structure of these hooks/apis to follow https://github.com/openedx/frontend-app-course-authoring/pull/645/ that is using useQuery after pulling it from master. https://github.com/openedx/frontend-app-course-authoring/commit/6e1dfc72e45f2f410927363a2794c0a30c5c38e0

Reply:
My bad, I think I got confused switching between PRs. Sorry about this, I think this looks good.